### PR TITLE
Formatting for register field logging

### PR DESCRIPTION
**Pull request by dscott90**

### CODE REVIEW

File: chipsec/logger.py [hunk 1]
```diff
@@ -395,6 +396,35 @@ def logger():
...
+        return tuple(min_col_widths)
+
+    # If list of strings
+    if type(str_table[0]) is str:
+        col_lengths = []
+        for string in str_table:
+            if len(string) < min_width:
+                col_lengths.append(min_width)
+            else:
+                col_lengths.append(len(string))
```
Comment (kevinsun49) [inline chipsec/logger.py:426]: Suggesting a more Pythonic approach with list comprehension (and also converting the if-else block into a max function):
`col_lengths = [max(min_width, len(string) for string in str_table]`

**Comment by dscott90** [comment]
WIP

File: chipsec/logger.py [hunk 2]
```diff
@@ -395,6 +396,32 @@ def logger():
...
+    column_widths = get_column_widths(clean_data)
+    min_widths = find_min_col_widths(column_widths)
+    return tuple(min_widths)
+
+def clean_data_table(data_table):
+    clean_table = [extract_column_values(row) for row in data_table]
+    return clean_table
+
+def extract_column_values(row_data):
+    clean_row = [row_data[0]]
```
Comment (npmitche) [inline chipsec/logger.py:410]: A cleaner way to do this would be: 
```suggestion
    clean_row = row_data[:1]
```

File: chipsec/logger.py [hunk 3]
```diff
@@ -395,6 +396,32 @@ def logger():
...
+    [clean_row.append(value) for value in additional_column_values]
+    return clean_row
+
+def get_column_widths(data):
+    col_widths = [[len(col) for col in row] for row in data]
+    return col_widths
+
+def find_min_col_widths(col_data, minimum_width = 2):
+    columns_per_row = len(col_data[0])
+    min_widths = ([(max(rows[i] for rows in col_data)) for i in range(columns_per_row)])
```
Comment (npmitche) [inline chipsec/logger.py:421]: Aren't you finding the max_widths? Should this variable/function be renamed?

File: chipsec/logger.py [hunk 4]
```diff
@@ -395,6 +396,32 @@ def logger():
     """Returns a Logger instance."""
     return _logger
 
+def minimum_column_widths(table_data):
```
Comment (npmitche) [inline chipsec/logger.py:399]: See comment on line 421

File: chipsec/logger.py [hunk 5]
```diff
@@ -27,8 +27,9 @@
 import binascii
 import sys
 import os
-from time import localtime, strftime
 import atexit
+from time import localtime, strftime
+from typing import Union, List, Tuple
```
Comment (npmitche) [inline chipsec/logger.py:32]: You're not using type hinting anymore. Recommend either adding type hinting or removing this import as it's not being used ATM.

**Review by npmitche — APPROVED** [review]

**Review by BrentHoltsclaw — APPROVED** [review]
It would be helpful to describe what the table data input is supposed to look like

**Comment by BrentHoltsclaw** [comment]
General comment, it may be helpful to split the formatting functionality and the logging functionality into separate files

**Pull request merged**